lib/gnat/consumer_supervisor.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Gnat.ConsumerSupervisor do
+  use GenServer
+  require Logger
+
+  def start_link(settings, options) do
+    GenServer.start_link(__MODULE__, settings, options)
+  end
+
+  def init(settings) do
+    Process.flag(:trap_exit, true)
+    {:ok, task_supervisor_pid} = Task.Supervisor.start_link()
+    connection_name = Map.get(settings, :connection_name)
+    subscription_topics = Map.get(settings, :subscription_topics)
+    consuming_function = Map.get(settings, :consuming_function)
+    send self(), :connect
+    state = %{
+      connection_name: connection_name,
+      connection_pid: nil,
+      consuming_function: consuming_function,
+      status: :disconnected,
+      subscription_topics: subscription_topics,
```

Comment:
I think I'm misunderstanding what you mean by sub topics. As far as the `Gnat` connection is concerned all topics and subscriptions are the exact same. This consumer code is just meant to be some helpers around automatically subscribing to topics that you want to keep your application listening for. So when the connection comes and goes you automatically re-create the long-lived topics.

Are you thinking that the connection should remember previous subscriptions and automatically re-subscribe to them when a new connection is formed? That would be part of the responsibility of the `Gnat` connection rather than a separate responsibility?